Extensions/retags.js

```diff
@@ -96,9 +96,9 @@ XKit.extensions.retags = {
 			}
 			if (url) {
 				url = url.split('/');
-				var host = url[2], id = url[4], cache = 'retags_'+id;
-				if (localStorage.getItem(cache) !== null) {
-					XKit.extensions.retags.append($t,cls,$c,JSON.parse(localStorage.getItem(cache)));
+				var host = url[2], id = url[4], cache = 'post_'+id;
+				if (XKit.storage.get('retags', cache, null) !== null) {
+					XKit.extensions.retags.append($t, cls, $c, XKit.storage.get('retags', cache));
```

Comment:
Hmm, are you sure? It looks like `XKit.storage.get_all()` does its own `JSON.parse()` call:
```javascript
		get_all: function(extension_id) {
			var m_data = XKit.tools.get_setting("xkit_extension_storage__" + extension_id, "");
			if (m_data !== "") {
				try {
					return JSON.parse(m_data);
				} catch(e) {
				}
			}
			return {};
		},
```
And `XKit.storage.get()` just calls to `get_all` and then fetches by key, so the JSON is already parsed at that point.

The code in this branch Works For Me in my XKit installation so I doubt it's failing to deserialise?